src/HDF5.jl

```diff
@@ -1371,10 +1373,10 @@ function read{S<:String}(obj::DatasetOrAttribute, ::Type{Array{S}})
         ret = Array{S}(sz)
         if isvar
             # Variable-length
-            buf = Vector{Ptr{UInt8}}(len)
+            buf = Vector{Cstring}(len)
             h5t_set_size(memtype_id, H5T_VARIABLE)
             readarray(obj, memtype_id, buf)
-            # FIXME? Who owns the memory for each string? Will Julia free it?
+            # FIXME? Who owns the memory for each string? Julia v0.6+ won't free it.
```

Comment:
Checking the docs, it appears there may supposed to be a call to `H5Dvlen_reclaim` before the call to `h5t_close` (and this function should perhaps be wrapped in try/catch so those get called even if there's an error with reading part of the data.